### PR TITLE
add examples/ethereum_implementation.md + update README

### PR DESCRIPTION
**Pull request by tboeckmann**
update README

### CODE REVIEW

File: Readme.MD [hunk 1]
```diff
@@ -3,32 +3,67 @@
 ![https://travis-ci.com/mailchain/mailchain](https://img.shields.io/travis/com/mailchain/mailchain.svg?style=for-the-badge)
 ![https://coveralls.io/github/mailchain/mailchain](https://img.shields.io/coveralls/github/mailchain/mailchain.svg?style=for-the-badge)
 
-Mailchain is an email-like application. Using blockchain encryption and decentralized storage it delivers a simple, secure, email-like experience to blockchain users. Using Mailchain, Ethereum users can now send and receive rich-media HTML messages with attachments via an Ethereum address.
+## About Mailchain ###
+Extract from the [Mailchain specification](https://github.com/mailchain/mailchain-specification):
```
Comment (robdefeo) [inline Readme.MD]: This is refered to later and just takes up space at the top.

File: Readme.MD [hunk 2]
```diff
@@ -3,32 +3,67 @@
...
+* A user sends a message. The message is encrypted and stored using the recipient’s public key so that only the recipient can locate and decrypt the message contents. The encrypted location of the message is sent by the sender to the recipient in the data field of a transaction.
+
+* To read a message, a recipient uses a corresponding private key to decrypt the location of the message from the data field of a transaction and decrypt the message contents.
+
+---
 
 ## Getting started
 
-Navigate to xx Download the client
+1. Navigate to https://github.com/mailchain/mailchain/releases to download te latest mailchain build.
```
Comment (robdefeo) [inline Readme.MD]: https://github.com/mailchain/mailchain/releases/latest takes you to the latest version

File: Readme.MD [hunk 3]
```diff
@@ -3,32 +3,67 @@
...
+*For example: In the case of Ethereum the data will be prefixed with 0x and added to the input field.* 
+
+An explaination of the Ethereum implementation can be count in [examples/ethereum_implementation.md](examples/ethereum_implementation.md) file.
 
 #### Format
 
-Chain data is formatted as follows `ChainData = encoded(ChainPrefix + Prefix + MultiFormatData + ChainSuffix)`. All values are stored are bytes that are then `encoded` to be stored on the chain. The `ChainPrefix` and `ChainSuffix` allow for any chain specific formatting to be applied.
+tx_data is formatted as follows `tx_data = encoded(ChainPrefix + Prefix + MultiFormatData + ChainSuffix)`. All values are stored are bytes that are then `encoded` to be stored on the chain. The `ChainPrefix` and `ChainSuffix` allow for any blockchain protocol-specific formatting to be applied.
```
Comment (robdefeo) [inline Readme.MD]: Thought we were getting rid of suffix?

Comment (robdefeo) [inline Readme.MD]: Thought we were getting rid of suffix?

File: examples/ethereum_implementation.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
+# Example Implementation in Ethereum
```
Comment (robdefeo) [inline examples/ethereum_implementation.md]: # Ethereum Implementation 
A good place for it to live would be docs/ethereum

File: examples/ethereum_implementation.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+    The file storage location is returned. E.g. https://mcx.mx/002c5d4ba47ce66f9e4b1f36f35e50c357aded81dfb9b98a89b8a80d5ca347b2a16f08dc5d37d255378ddcf3380d-220426516c9b
+    
+    [returns `message_location`]
+
+1. The `message_location` is encrypted using the recipient public key (default AES-256-CBC). This is a byte array.
+
+    [returns `encrypted_message_location`]
+
+    1. A protocol buffer (or protobuff) (https://developers.google.com/protocol-buffers) is created containing the following fields:
```
Comment (robdefeo) [inline examples/ethereum_implementation.md]: protobuff is with a single 'f' protobuf. There are multiple examples

File: examples/ethereum_implementation.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+1. In the web interface, a user composes a message, including the following fields:
+
+    Field | Description | Example
+    | - | - | - |
+    To: | The recipient public address | `0xd5ab4ce3605cd590db609b6b5c8901fdb2ef7fe6`
+    From: | The sender public address | `0x92d8f10248c6a3953cc3692a894655ad05d61efb`
+    Reply-To: | The public address responses should be sent to | `0x92d8f10248c6a3953cc3692a894655ad05d61efb`
+    Subject: | The message subject | 
+    Body: | The message body |
+    Public Key: | The recipient public key^ | `0x69d908510e355beb1d5bf2df8129e5b6401e1969891e8016a0b2300739bbb00687055e5924a2fd8dd35f069dc14d8147aa11c1f7e2f271573487e1beeb2be9d0`
```
Comment (robdefeo) [inline examples/ethereum_implementation.md]: ^ looks eroneous

File: examples/ethereum_implementation.md (hunk 4 above)
Comment (robdefeo) [reply examples/ethereum_implementation.md]: Actually I think `internal/pkg/chains/ethereum/readme.MD` is a better place

File: Readme.MD (hunk 3 above)
Comment (tboeckmann) [reply Readme.MD]: Yes.
Has this gone from the the rest of the repo?
Once it has, I'll remove refs everywhere:

[] this README
[] Ethereum Implementation
[] Mailchain specification

File: examples/ethereum_implementation.md (hunk 5 above)
Comment (tboeckmann) [reply examples/ethereum_implementation.md]: Changed everywhere.

File: examples/ethereum_implementation.md (hunk 6 above)
Comment (tboeckmann) [reply examples/ethereum_implementation.md]: clarified the purpose of the note.

File: examples/ethereum_implementation.md (hunk 4 above)
Comment (tboeckmann) [reply examples/ethereum_implementation.md]: Could be- thanks for clarifying
Moved to this place.

File: Readme.MD (hunk 1 above)
Comment (tboeckmann) [reply Readme.MD]: Sorted.

**Pull request merged**